### PR TITLE
Fix serialization of `Nut` model

### PR DESCRIPTION
**Pull request by kirillzh**
A serialization unit test was failing with
```
Unexpected JSON token at offset 170: Expected start of the array '[', but had '[' instead at path: $.nuts['4'].methods[0]
```

because response structure is different from the one used by `Nut`.

Fix by introducing a `Method` model type as part of a `Nul` type matching actual json response structure.

### CODE REVIEW

**Review by thunderbiscuit — APPROVED** [review]
ACK e15cc03852ac82d2cb26674afd105c6771290464, thanks for the fix.

This mint info endpoint needs work to be typesafe, particularly as this `Method` object is really defined differently for different NUTs, but that's work for future us. Glad to see the tests fixed.

**Comment by thunderbiscuit** [comment]
@kirillzh would you mind rebasing before I merge?

**Comment by kirillzh** [comment]
> @kirillzh would you mind rebasing before I merge?

Doesn't look like there are any changes to rebase (tried locally)

**Comment by thunderbiscuit** [comment]
Yeah sorry I just try to not merge any merge commits, so rebases are required, but if I let GitHub do the rebase through the big green merge/rebase button, I lose the GPG signature on your commit. So either people need to rebase themselves or I do the rebase, but then it adds me as an author to the commit, so I often let people rebase.

**Pull request merged**
